src/assets/js/dynamic-addon-cards.js

```diff
@@ -0,0 +1,46 @@
+/* eslint no-console: 0 */
+/* global document, fetch */
+(function dynamicAddonCards() {
+  const AMO_BASE_URL = 'https://addons.mozilla.org';
+
+  // `StaticAddonCard` elements can be marked as unavailable by adding a
+  // special CSS class. The style as well as the content is already embedded.
+  const convertToUnavailableCard = (card) => {
+    card.classList.add('StaticAddonCard--is-unavailable');
+  };
+
+  const updateAddonCard = async (card) => {
+    const { addonId } = card.dataset;
+
+    if (!addonId) {
+      console.debug(`found an add-on card without an addonId`);
+      return;
```

Comment:
If this scenario occurred what would the end state look like? I'm wondering if this should be the same as unavailable instead of the early return.

Reply:
> If this scenario occurred what would the end state look like? 

I cannot see how this could happen tbh, both the class name and the data attribute are generated at the same time by the same code in the same function. I suppose if someone updates the `blog-utils` library and remove the data attribute, then that could happen but it shouldn't be possible (there is test coverage for the data attribute in addons-frontend).

That being said...

> I'm wondering if this should be the same as unavailable instead of the early return.

Yeah, it's probably easier to only have a unique way to handle errors so I'll do that.